src/linear_algebra/affine_space/barycentric_coords.lean

```diff
@@ -107,3 +107,17 @@ begin
   simp only [barycentric_coord_apply, hi, finset.affine_combination_eq_linear_combination, if_false,
     hw, mul_boole, function.comp_app, smul_eq_mul, s.sum_ite_eq, s.map_affine_combination p w hw],
 end
+
+@[simp] lemma barycentric_coord_eq_one_of_subsingleton [subsingleton ι] (q : P) (i : ι) :
+  barycentric_coord h_ind h_tot i q = 1 :=
+begin
+  have hp : (range p).subsingleton,
+  { rw ← image_univ,
+    apply subsingleton.image,
+    apply subsingleton_of_subsingleton, },
+  haveI := affine_subspace.subsingleton_of_subsingleton_span_eq_top hp h_tot,
+  let s : finset ι := {i},
+  have hi : i ∈ s, { simp, },
+  have hw : s.sum (function.const ι (1 : k)) = 1, { simp, },
+  convert barycentric_coord_apply_combination_of_mem h_ind h_tot hi hw,
```

Comment:
Finishing with `convert` is really a bad omen. Is there some `simp` lemma missing?

Reply:
Yeah that was a evil. I have added 752843cfec09006e2680239075772a3fe4acd303 to fix this.